src/main/kotlin/dev/supergrecko/kllvm/core/values/IntValue.kt

```diff
@@ -14,4 +18,18 @@ public class IntValue(llvmValue: LLVMValueRef) : Value(llvmValue) {
         return LLVM.LLVMConstIntGetSExtValue(llvmValue)
     }
     //endregion Core::Values::Constants::ScalarConstants
+
+    companion object {
+        //region Core::Values::Constants::ScalarConstants
+        public fun new(type: Type, value: Long, signExtend: Boolean): IntValue {
+            return IntValue(
+                LLVM.LLVMConstInt(
+                    type.getUnderlyingReference(),
+                    value,
+                    signExtend.toInt()
+                )
+            )
+        }
+        //endregion Core::Values::Constants::ScalarConstants
+    }
```

Comment:
Another thing we could potentially do is to introduce secondary constructors and drop the entire .new thing. Might be an even nicer solution

Reply:
One issue I came across while testing out secondary constructors is converting [things like this](https://github.com/superbytes/kllvm/blob/master/src/main/kotlin/dev/supergrecko/kllvm/core/types/IntType.kt#L19-L31) to use secondary constructors.

Perhaps a combination of the companion object and secondary constructors works?

Reply:
I didn't see the implementation in Types so I thought it didn't exist 😅.

That being said, I like the idea of keeping the factories within the class that we're constructing From an IDE discoverability perspective, it seems like a good thing. like, how do I get an instance of IntValue? Just type IntValue. and I'll get a bunch of auto completions in the IDE for factory functions. Or use a constructor (more on that later)

On the other hand, compatibility with the LLVM api might also be useful so that any one familiar with LLVM c++ api might have an easier time finding things.

Regarding the secondary constructor issue, yeah I think it's a really good idea. Constructors are supposed to construct and all.

Reply:
Should I remove this factory then?

Reply:
I'll leave it for now and merge. We're going to stick with the secondary constructors for functions which would overload the function. For things like [this](https://github.com/superbytes/kllvm/blob/master/src/main/kotlin/dev/supergrecko/kllvm/core/types/IntType.kt#L19-L31) we'll use the companion object.

I'll be performing the refactor after I've merged this and tag you for review